src/test/groovy/graphql/InterfacesImplementingInterfacesTest.groovy

```diff
@@ -0,0 +1,861 @@
+            }
+            
+            type Query {
+               mock: String
+            }
+            """
+
+        parseSchema(schema)
+
+        then:
+        def error = thrown(SchemaProblem)
+        error.errors.size() == 4
+
+        assertErrorMessage(error, "The interface type 'BaseInterface' [@n:n] field 'fieldA' does not have the same number of arguments as specified via interface 'InterfaceType' [@n:n]")
+        assertErrorMessage(error, "The interface type 'BaseInterface' [@n:n] has tried to redefine field 'fieldB' arguments defined via interface 'InterfaceType' [@n:n] from 'arg1:String =\"defaultVal\"' to 'arg1:String =\"defaultValX\"")
+        assertErrorMessage(error, "The interface type 'BaseInterface' [@n:n] has tried to redefine field 'fieldB' arguments defined via interface 'InterfaceType' [@n:n] from 'arg2:String' to 'arg2:String!")
+        assertErrorMessage(error, "The interface type 'BaseInterface' [@n:n] has tried to redefine field 'fieldB' arguments defined via interface 'InterfaceType' [@n:n] from 'arg3:Int' to 'arg3:String")
+    }
+
+    def 'When interface implements interface via extension and misses field arguments, then parsing fails'() {
+        when:
+        def schema = """
+            interface InterfaceType {
+                fieldA(arg1 : Int) : Int 
+                fieldB(arg1 : String = "defaultVal", arg2 : String, arg3 : Int) : String 
+            }
+
+            interface BaseInterface {
+                fieldX : Int
+                fieldA : Int
+                fieldB(arg1 : String = "defaultValX", arg2 : String!, arg3 : String) : String 
+            }
+            
+            extend interface BaseInterface implements InterfaceType
+            
+            type BaseType {
+                id: ID!
+            }
+
+            type Query {
+               mock: String
+            }
+            """
+
+        parseSchema(schema)
+
+        then:
+        def error = thrown(SchemaProblem)
+        error.errors.size() == 4
+
+        assertErrorMessage(error, "The interface extension type 'BaseInterface' [@n:n] field 'fieldA' does not have the same number of arguments as specified via interface 'InterfaceType' [@n:n]")
+        assertErrorMessage(error, "The interface extension type 'BaseInterface' [@n:n] has tried to redefine field 'fieldB' arguments defined via interface 'InterfaceType' [@n:n] from 'arg1:String =\"defaultVal\"' to 'arg1:String =\"defaultValX\"")
+        assertErrorMessage(error, "The interface extension type 'BaseInterface' [@n:n] has tried to redefine field 'fieldB' arguments defined via interface 'InterfaceType' [@n:n] from 'arg2:String' to 'arg2:String!")
+        assertErrorMessage(error, "The interface extension type 'BaseInterface' [@n:n] has tried to redefine field 'fieldB' arguments defined via interface 'InterfaceType' [@n:n] from 'arg3:Int' to 'arg3:String")
+    }
+
+    def assertErrorMessage(SchemaProblem error, expectedMessage) {
```

Comment:
Nice one!  We should do this in other tests.

Can you please rewrite all our tests to this pattern please! ;)